texastribune/donations
# CSS reorg round 2

#### What's this PR do?


- Removes duplicate element resets and restated classes
- Removes explicit CSS property assignments when they can naturally inherited (ie font family, color, etc)
- Moves all membership drive css into new components:  `6-components/thermometer` and `6-components/smd`
- Creates a `.splash` component with updated BEM-ifying and repurposes that for both the /donate and /blastform headers
- Remove use of `@extend` for fonts in favor of `@mixin`
- Brings all remaining views into the new CSS ecosystem

#### Why are we doing this? How does it help us?

We're trimming fat in all the outputted files. This will put us in a good place to start thinking of our templates as modular UI elements

#### How should this be manually tested?

Visit every possible view of the /donations app and check for any style inconsistencies. Nothing should be wildly different from what's live.

#### How should this change be communicated to end users?
N/A

#### Are there any smells or added technical debt to note?
We're left with a bunch of files; mass deletion round of those to come.

#### What are the relevant tickets?

#### Have you done the following, if applicable:
***(optional: add explanation between parentheses)***

* [ ] Added automated tests? *( N/A )*
* [ ] Tested manually on mobile? *( will after PR deploy)*
* [ ] Checked for security implications? *( N/A )*
* [ ] Updated the documentation/wiki? *( N/A )*

#### TODOs / next steps:

* [x] Get business and circle on the same CSS base (remove dup CSS)
* [x] Integrate blastform (/blastform) into new structure - create components where needed
* [x] Integrate homepage (/donate) into new structure - create components where needed
* [x] Replace silent classes with mixins
* [ ] Turn one-off page styles into reusable components (example circle-wall)
* [ ] Delete legacy CSS*
* [ ] Remove opinionated sizing on heading tags
* [ ] Convert class names to use UI namespacing c-component o-object
* [ ] Scope out adding our own spin to ITCSS with a typography folder (maybe after objects/before components)
* [ ] Clean up objects modules. Remove specifically-styled files like borders and move those into to components
